activerecord/test/cases/calculations_test.rb

```diff
@@ -67,6 +67,14 @@ def test_should_return_float_average_if_db_returns_such
     assert_equal 37.5, value
   end
 
+  def test_should_return_decimal_average_if_db_returns_such
+    NumericData.create!(decimal_number: 37.5)
+
+    value = NumericData.average(:decimal_number)
+    assert_instance_of BigDecimal, value
```

Comment:
This test is failing for sqlite3 and postgresql. I removed it, but it seems things are now broken. Can you take a look?

Reply:
Working on it here: https://github.com/rails/rails/pull/41418